nova/virt/vmwareapi/driver.py

```diff
@@ -487,6 +488,30 @@ def get_inventory(self, nodename):
                 'max_unit': stats['mem']['max_mem_mb_per_host'],
                 'step_size': 1,
             }})
+
+            # Calculate total reservable memory, depending on either the
+            # configured tolerance for failed hypervisors or, if that setting
+            # is disabled, a ratio of maximum reserved memory to total
+            # available memory.
+            memory_available_total = stats['mem']['total'] - reserved_memory_mb
+            hv_count = stats.get('hypervisor_count', False)
+            max_fail_hvs = \
+                max(0, CONF.vmware.memory_reservation_cluster_hosts_max_fail)
+            if max_fail_hvs and hv_count:
+                total_memory_reservable = \
+                    memory_available_total * (1 - max_fail_hvs / hv_count)
+            else:
+                total_memory_reservable = memory_available_total \
+                    * CONF.vmware.memory_reservation_max_ratio_fallback
+            result.update({
+                utils.MEMORY_RESERVABLE_MB_RESOURCE: {
+                    'total': int(total_memory_reservable),
+                    'reserved': 0,  # n.b. This does not refer to VM-reserved
+                                    # memory, but host-reserved memory! -> zero
```

Comment:
Do you mean `stats['mem']['total'] - (total_memory_reservable + reserved_memory_mb)`?

Reply:
The "host" in the comment is more meant as a type name, rather than "one host" -- similar to "VM-reserved memory".

It's probably going away anyway, since I kind of like the idea of exposing more information to other parties. But I still have to think about it...

Reply:
Turned this into the total-and-reserved style you described, and it seems good to me.